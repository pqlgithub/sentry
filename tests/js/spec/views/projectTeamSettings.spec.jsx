import React from 'react';
import {shallow} from 'enzyme';

import {Client} from 'app/api';
import ProjectTeams from 'app/views/settings/project/projectTeams';

describe('ProjectTeamsSettings', function() {
  let org;
  let project;
  let team;

  beforeEach(function() {
    org = TestStubs.Organization();
    project = TestStubs.Project();
    team = TestStubs.Team();

    Client.addMockResponse({
      url: `/projects/${org.slug}/${project.slug}/`,
      method: 'GET',
      body: project,
    });
    Client.addMockResponse({
      url: `/projects/${org.slug}/${project.slug}/teams/`,
      method: 'GET',
      body: [team],
    });
  });

  describe('render()', function() {
    it('renders', function() {
      let wrapper = shallow(
        <ProjectTeams
          params={{orgId: org.slug, projectId: project.slug}}
          organization={org}
        />,
        {
          context: {
            router: TestStubs.router(),
          },
        }
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
