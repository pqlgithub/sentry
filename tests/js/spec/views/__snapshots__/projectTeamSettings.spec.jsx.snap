// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectTeamsSettings render() renders 1`] = `
<SideEffect(DocumentTitle)
  title="Sentry"
>
  <div>
    <SettingsPageHeading
      title="Teams"
    />
    <div
      className="panel panel-default horizontal-scroll"
    >
      <table
        className="table"
      >
        <thead>
          <tr>
            <th>
              Team
            </th>
            <th
              style={
                Object {
                  "width": 120,
                }
              }
            />
          </tr>
        </thead>
        <tbody>
          <TeamRow
            access={
              Set {
                "org:read",
                "org:write",
                "org:admin",
                "project:read",
                "project:write",
                "project:admin",
                "team:read",
                "team:write",
                "team:admin",
              }
            }
            key="1"
            onRemove={[Function]}
            orgId="org-slug"
            projectId="project-slug"
            team={
              Object {
                "id": "1",
                "name": "Team Name",
                "slug": "team-slug",
              }
            }
          />
        </tbody>
      </table>
    </div>
  </div>
</SideEffect(DocumentTitle)>
`;
